lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -444,8 +444,14 @@ public function commit($entity = null)
 
             // Entity deletions come last and need to be in reverse commit order
             if ($this->entityDeletions) {
-                for ($count = count($commitOrder), $i = $count - 1; $i >= 0 && $this->entityDeletions; --$i) {
-                    $this->executeDeletions($commitOrder[$i]);
+                // There are some cases in which the original $commitOrder is not correct for deletions,
+                // so we need to call a particular function to compute the right order.
+                // For example see tests/Doctrine/Tests/ORM/Functional/Ticket/GH9376Test.php.
+                // Without computing the special order for deletions, it tries to delete the entity set as a foreign key in another entity.
+                // getCommitOrderForDeletions() is similar to getCommitOrder(), but it operates only with entities set to be deleted.
```

Comment:
This should be part of the description of a commit message. Your method name seem to describe well what is happening.

Reply:
fixed